qiskit/pulse/model/frames.py

```diff
@@ -0,0 +1,197 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Frames
+"""
+
+from abc import ABCMeta
+from typing import Optional
+
+import numpy as np
+
+from qiskit.pulse.exceptions import PulseError
+
+
+class Frame(metaclass=ABCMeta):
+    """Base class for pulse module frame.
+
+    Because pulses used in Quantum HW are typically AC pulses, the carrier frequency and phase
+    must be defined. The :class:`Frame` is the object which sets the frequency and phase for the carrier,
+    and each pulse, and most instructions are associated with a frame. The different types of frames
```

Comment:
The punctuation is confusing. Do you mean this?

```suggestion
    and each pulse and most other instructions are associated with a frame. The different types of frames
```